src/main/com/example011/radek/v1/Mines.java

```diff
@@ -0,0 +1,173 @@
+package main.com.example011.radek.v1;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Scanner;
+
+public class Mines {
+
+    protected int playgroundSize;
+
+    protected int numMines;
+
+    protected String[][] playground;
+
+    public Mines(int size, int mines) {
+
+        if (setPlaygroundSize(size)) {
+            int setNumMinesResult = setNumMines(mines);
+            if (setNumMinesResult == 1) {
+                System.out.println("You can't have more mines, then you have playground size !");
+                System.exit(0);
+            } else if (setNumMinesResult == 2) {
+                System.out.println("To place mine, you need playground of minimum size of 10 boxes!");
+                System.exit(0);
+            }
+        } else {
+            System.out.println("Bad size of playground, size must be bigger then 5!");
+            System.exit(0);
+        }
+    }
+
+    public void fillPlaygroundWithMines() throws NoSuchAlgorithmException {
+
+        playground = new String[playgroundSize][playgroundSize];
+        SecureRandom random = SecureRandom.getInstanceStrong();
+        int placedMines = 0;
+        while (placedMines != getNumMines() || placedMines == 0) {
+            int generatedRowNumber = random.nextInt(playgroundSize);
+            int generatedColumnNumber = random.nextInt(playgroundSize);
+            if (playground[generatedRowNumber][generatedColumnNumber] == null) {
+                playground[generatedRowNumber][generatedColumnNumber] = "x";
+                placedMines = placedMines + 1;
+            }
+        }
+    }
+
+    public void printPlayground(boolean showMines) throws NoSuchAlgorithmException {
+
+        System.out.println("Generating playground...");
+        String printMessage = "";
+        for (int i = 0; i < playgroundSize; i++) {
+            if (i > 0) {
+                printMessage = printMessage + "\n";
```

Comment:
Strnig nikdy nespojuj pres + ... v malem mnozstvi jako tu by to nevadilo, ale nejlepe se to ani neuc ... pouzij stringbuilder

Reply:
Ok, opraveno